### ISSUE TITLE
[BUG] yflip and lens issues

### ISSUE THREAD

**Issue by RibeiroAndre** [JuliaPlots/Plots.jl#3843]
## Details
Trying to use lens with yflip is giving me strange results. In one case, the grey line connecting the lens to its box is in the wrong place. In another case, the original plot gets unflipped.
```
using Plots
plotlyjs()
plot(rand(100),rand(100),yflip=true)
lens!([0,.2], [0,.2], inset = (1, bbox(0.35, 0.0, 0.4, 0.4)), yflip=true)
```

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |     |     |x
pyplot       |     |     |x
plotlyjs     |  x   |     |
pgfplotsx    |     |     |x
unicodeplots |     |     |x
inspectdr    |     |     |x
gaston       |     |     |x

### Versions

[f0f68f2c] PlotlyJS v0.18.7
[91a5bcdd] Plots v1.21.3

Version info
Julia Version 1.6.2
Commit 1b93d53fc4 (2021-07-14 15:36 UTC)
Platform Info:
OS: Windows (x86_64-w64-mingw32)
CPU: Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz
WORD_SIZE: 64
LIBM: libopenlibm
LLVM: libLLVM-11.0.1 (ORCJIT, skylake)

**Comment by RibeiroAndre** [comment]
I don't believe this is fixed. Testing it now gives a very different behavior. The yflip seems to apply to the lens, but when I plot the lens, the original plot gets y-unflipped.
Not sure how to reopen the ticket.

**Comment by BeastyBlacksmith** [comment]
I can reproduce this for plotlyjs, but it works correctly for gr
